packages/core/src/Stores/contract-store.js

```diff
@@ -241,8 +252,22 @@ export default class ContractStore extends BaseStore {
             this.barriers_array = this.createBarriersArray(contract_info, is_dark_mode);
             return;
         }
+        if (
+            this.barriers_array.length === 1 &&
+            isResetContract(contract_info.contract_type) &&
+            contract_info.reset_time
```

Comment:
Field `reset_time` will appear after half of contract is gone. When FE starts to receive it, we should draw 2nd barrier (reset barrier)